ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/KnownPeers.hs

```diff
@@ -20,207 +21,291 @@ import           Control.Monad.Class.MonadTimer
 
 import qualified Ouroboros.Network.PeerSelection.EstablishedPeers as EstablishedPeers
 import           Ouroboros.Network.PeerSelection.Governor.Types
+import           Ouroboros.Network.PeerSelection.KnownPeers (isKnownLedgerPeer)
 import qualified Ouroboros.Network.PeerSelection.KnownPeers as KnownPeers
+import           Ouroboros.Network.PeerSelection.LedgerPeers (IsLedgerPeer (..))
 import qualified Ouroboros.Network.PeerSelection.LocalRootPeers as LocalRootPeers
+import           Ouroboros.Network.PeerSelection.PeerAdvertise.Type
+                     (PeerAdvertise (..))
+import           Ouroboros.Network.PeerSelection.PeerSharing.Type
+                     (PeerSharing (..))
+import           Ouroboros.Network.Protocol.PeerSharing.Type (PeerSharingAmount)
 
 
 ---------------------------
 -- Known peers below target
 --
 
 
--- | If we are below the target of /known peers/ we gossip (if we are above the
--- gossip request threashold).
+-- | If we are below the target of /known peers/ we peer share (if we are above
+-- the peer share request threashold).
 --
 belowTarget :: (MonadAsync m, MonadTimer m, Ord peeraddr)
```

Comment:
Should't we move this either to `Ouroboros.Network.PeerSelection.Governor.EstablishedPeers` or its' own `Ouroboros.Network.PeerSelection.Governor.PeerSharing`.  We no longer do peer sharing with cold peers, but with established ones and thus this being in `KnownPeers` might be a bit confusing.

Reply:
I get your point however this is a method that is called when we are below the known peer target so I think it is a good place for it to be. The other files for Established and Active peers all have methods like this regarding the targets. So in terms of mental model I think it fits here 😛

Reply:
The same applies to `RootPeers`. We will also have `Ouroboros.Network.PeerSelection.Governor.KnownBigLedgerPeers`.  So why shared peers should be special in that regard?

I think eventually we should have:
* `Ouroboros.Network.PeerSelection.Governor.KnownPeers.RootPeers`
* `Ouroboros.Network.PeerSelection.Governor.KnownPeers.BigLedgerPeers`
* `Ouroboros.Network.PeerSelection.Governor.KnownPeers.SharedPeers`

Reply:
We can do that later.

Reply:
Got it :)